### PR TITLE
8244810: [lworld] Decommission arrayStorageProperties from Klass:array_klass() et el

### PR DESCRIPTION
**Pull request by MrSimms**
First part of LW2 to LW3 arrays

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8244810](https://bugs.openjdk.java.net/browse/JDK-8244810): [lworld] Decommission arrayStorageProperties from Klass:array_klass() et el

### Download
`$ git fetch https://git.openjdk.java.net/valhalla pull/44/head:pull/44`
`$ git checkout pull/44`

### CODE REVIEW

**Comment by MrSimms** [comment]
Partially backing out of 8222525 (https://github.com/openjdk/valhalla/commit/f49762698cfc350280eca28cd3daf67f0a000e9f)

File: src/hotspot/share/oops/arrayStorageProperties.hpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2019, 2020, Oracle and/or its affiliates. All rights reserved.
```
Comment (fparain) [inline src/hotspot/share/oops/arrayStorageProperties.hpp:2]: (Comment for the whole file)
I understand this is the first stage of a set of complex changes to remove the `ArrayStorageProperties` code, but cleaning up the `ArrayStorageProperties` as things are getting removed might help to see progress (and would make reviews easier).
For instance:
  - `clear_flattened()`
  - `set_flattened()`
  - `set_null_free()`
  - `for_signature(Symbol* sig)`
seem to not be used anymore and could be removed.

File: src/hotspot/share/memory/oopFactory.cpp [hunk 2]
```diff
@@ -152,7 +152,7 @@ arrayOop oopFactory::new_valueArray(Klass* klass, int length, TRAPS) {
 objArrayHandle oopFactory::copy_valueArray_to_objArray(valueArrayHandle array, TRAPS) {
   int len = array->length();
   ValueArrayKlass* vak = ValueArrayKlass::cast(array->klass());
-  objArrayHandle oarray = new_objArray_handle(vak->element_klass(),
+  objArrayHandle oarray = new_objArray_handle(vak->element_klass()->super(),
```
[the remark this answers is not in the document]
Comment (MrSimms) [reply src/hotspot/share/memory/oopFactory.cpp:155]: Totally agree.

File: src/hotspot/share/oops/arrayStorageProperties.hpp (hunk 1 above)
Comment (MrSimms) [reply src/hotspot/share/oops/arrayStorageProperties.hpp:2]: Correct, ultimately the whole thing can go...but yes, they can go now.

**Pull request closed without merging**